pkg/bpf/bpffs.go

```diff
@@ -105,6 +106,19 @@ func OpenAfterMount(m *Map) error {
 	return nil
 }
 
+//IsBpffs check if the path is a valid bpf filesystem
+func IsBpffs(path string) bool {
+	// This is the value of the BPF Filesystem. If is into the container the
+	// mountpoint doesn't provide enough information. Defined on uapi/linux/magic.h
+	magic := uint32(0xCAFE4A11)
+	var fsdata unix.Statfs_t
+	if err := unix.Statfs(path, &fsdata); err != nil {
+		log.Errorf("%s is not mounted", path)
+		return false
+	}
+	return int32(magic) == int32(fsdata.Type)
```

Comment:
We need a descriptive error message here or below that says that the file system is mounted but is not a BPF filesystem. You could just `log.Fatal()` here with a descriptive error which will automatically quit the agent.

Reply:
Hey @tgraf 

Yep, but the `log.Fatal()` is on `MountFS` function. I made this due `IsBpffs` is only for checking if it's correct, and in `MountFS` is the place where the actions are executed.

Reply:
That's fine. I just want it to be crystal clear to users when something is wrong. Right now there is no indication that the filesystem type is wrong. It will just say:
> BPFfs is not mounted. Check BPF mount on ...

The user will run `mount` and see a mountpoint at `/sys/fs/bpf` and get all confused.

Reply:
Should be ok now. Thanks!

Reply:
Did you forget to push the changes? I'm not seeing an updated error message.

Reply:
https://github.com/cilium/cilium/pull/1460/files#diff-b30aa362679bf44aaf221fe7cc2dc5a4R134

Reply:
OK, thanks